kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -48,9 +48,32 @@ impl<'tcx> GotocCtx<'tcx> {
                         .goto_expr
                         .assign(self.codegen_rvalue(r, location).cast_to(Type::c_bool()), location)
                 } else {
-                    unwrap_or_return_codegen_unimplemented_stmt!(self, self.codegen_place(l))
+                    match r {
```

Comment:
But this would no longer solve the problem: I need to modify the left-hand side expression (well, I should rather say: I'd prefer to modify the left-hand side expression, else I need to express the right-hand side as a byte-update expression, which so far hasn't been introduced in Kani).

Reply:
... but then your proposed implementation would do the trick and we could get rid of this hack.